lib/ex_unit/lib/ex_unit/diff.ex

```diff
@@ -1145,6 +1145,9 @@ defmodule ExUnit.Diff do
   # We escape container types to make a distinction between AST
   # and values that should be inspected. All other values have no
   # special AST representation, so we can keep them as is.
+  # Maps should be formatted not inspected.
+  defp escape(other) when is_struct(other), do: other
+  defp escape(elem) when is_map(elem), do: {:%{}, [], Map.to_list(elem)}
```

Comment:
I could wrap it in a tuple and then have a custom function for it in safe_to_algebra